client/app/scripts/superdesk-dictionaries/dictionaries.js

```diff
@@ -63,7 +63,15 @@
         };
 
         this.update = function update(dictionary, data, success, error) {
-            return api.save('dictionaries', dictionary, data).then(success, error);
+            var sendData = {};
+            angular.forEach(data, function(val, key) {
+                if (key !== 'content') {
+                    sendData[key] = val === null? val: val.toString();
+                } else {
+                    sendData[key] = val;
+                }
```

Comment:
IMO your proposal is not so readable because the statement is too long and with if statement is more easier to  understood to code.